### ISSUE TITLE
Use the full AST in MessageContext

### ISSUE THREAD

**Issue by stasm** [blushingpenguin/Fluent.Net#2]
The runtime AST is a perf optimization designed specifically with the JavaScript frontend in mind. Do you think C# needs it as well?

For reference, we're actually thinking of dropping it in favor of the full AST, provided the runtime parser can keep its current performance: https://github.com/projectfluent/fluent.js/issues/220.

The Python implementation of `MessageContext` (currently a WIP) also uses the full AST: https://github.com/projectfluent/python-fluent/pull/67.

**Comment by blushingpenguin** [comment]
No, I don't think it needs it, it actually doesn't make a lot of sense in C#.  I will eventually change it over to work with the full AST as you propose (however given it's effectively an implementation detail then it shouldn't affect _using_ the library).  Thanks for the suggestion!
